tests/data/test_types.py

```diff
@@ -179,20 +194,24 @@ def test_traj_unequal_to_perturbations(
             assert trajectory != types.Trajectory(
                 obs=trajectory.obs[: new_length + 1],
                 acts=trajectory.acts[:new_length],
-                infos=trajectory.obs[:new_length],
+                infos=trajectory.infos[:new_length]
+                if trajectory.infos is not None
+                else None,
                 terminal=trajectory.terminal,
             )
 
         # Or with contents changed
         for t in [trajectory, trajectory_rew]:
-            as_dict = dataclasses.asdict(t)
+            as_dict = types.dataclass_quick_asdict(t)
             for k in as_dict.keys():
                 perturbed = dict(as_dict)
                 if k == "infos":
                     perturbed["infos"] = [{"foo": 42}] * len(as_dict["infos"])
+                elif isinstance(as_dict[k], types.DictObs):
+                    perturbed[k] = as_dict[k].map_arrays(lambda x: x + 1)
                 else:
                     perturbed[k] = as_dict[k] + 1
-                assert trajectory != type(t)(**perturbed)
+                assert t != type(t)(**perturbed)
```

Comment:
Unrelated minor bug fix. Previously would always return false for `t = trajectory_rew` even without perturbations